yew-macro/tests/macro/classes-pass.rs

```diff
@@ -0,0 +1,25 @@
+use yew::prelude::*;
+
+fn compile_pass() {
+    classes!("one", "two");
+    classes!("one");
+    classes!();
+
+    classes!(vec!["one", "two"]);
+    classes!(vec!["one"]);
+    classes!(Vec::<&'static str>::new());
+    classes!(vec!["one"], vec!["two"]);
+
+    let some = Some("one");
+    let none: Option<&'static str> = None;
+    classes!(some, none);
+
+    let string = "one".to_string();
+    classes!(string.clone());
+    classes!(string, "two", vec!["three"]);
+
+    let array = ["one", "two"];
+    classes!(&array[..]);
+}
```

Comment:
This is where the `#![no_implicit_prelude]` attribute comes in. It helps us make sure that the macro doesn't accidentally use an item from the local scope. In order for it to be effective we also have to avoid bringing anything into scope. That makes things slightly more verbose, but proper macro hygiene is more important.

```suggestion
#![no_implicit_prelude]

fn compile_pass() {
    // multiple literals
    ::yew::classes!("one", "two");
    // single literal
    ::yew::classes!("one");
    // empty
    ::yew::classes!();

    // multiple expressions
    ::yew::classes!(::std::vec!["one"], ::std::vec!["two"]);
    // single expression
    ::yew::classes!(::std::vec!["one", "two"]);

    // optional classes
    ::yew::classes!(
        ::std::option::Option::Some("one"),
        ::std::option::Option::None::<&'static str>
    );

    // mixed types
    ::yew::classes!("one".to_owned(), "two", ::std::vec!["three"]);
}
```

I also...
- added comments to explain what each case is testing for
- removed some duplicate cases (as in, they weren't testing any specific feature of the macro). We only need to test the macro here and because it heavily relies on `Classes` we can just that separately